subsys/bluetooth/host/iso.c

```diff
@@ -1152,7 +1152,16 @@ void bt_iso_recv(struct bt_conn *conn, struct net_buf *buf, uint8_t flags)
 		pkt_seq_no = sys_le16_to_cpu(hdr->sn);
 		iso_info(buf)->sn = pkt_seq_no;
 
-		/* TODO: Drop the packet if NOP? */
+		if (flags == BT_ISO_DATA_VALID) {
```

Comment:
Do we need this conversion? It is a 2 bit field, hence we could simply align the API values with the values in the core spec - "All other values" covers only  the value 0b11.
enum {
	/** Convenience macro for when no PHY is set. */
	BT_ISO_FLAGS_VALID = 0,
	BT_ISO_FLAGS_CORRUPTED = 1,
	BT_ISO_FLAGS_LOST = 2,
	BT_ISO_FLAG_RFU
};
Thereby having less confusion when looking up the value in the BT specification.

Reply:
Not necessarily. The reason why I am doing this conversion is to handle the case where someone adds a (Zephyr specific) packet flag and possibly changes the values of this enum (so basically future proofing). 

An alternative is to add a built-time check (`BUILD_ASSERT`) that checks if the values of this enum is the same as the values from the HCI spec. 

Since the values are currently static and equal , then perhaps the compiler is able to optimize this part away anyhow.

Reply:
Agree